src/mapGenerator/mapgenManager.py

```diff
@@ -0,0 +1,156 @@
+        self.previousMaps = Settings.get('mapGenerator/deleteMaps')
+        self.generatorPath = os.path.join(fafpath.get_libdir(), "map-generator")
+        self.mapsFolder = getUserMapsFolder()
+
+        if self.previousMaps:
+            self.deletePreviousMaps()
+
+    def generateMap(self, mapname):
+        dataString =  mapname.split("neroxis_map_generator",1)[1]
+        version = re.search(r'\_(.*?)\_',dataString).group(1)
+        seed = dataString.rsplit('_', 1)[1]
+
+        actualPath = self.versionController(version)
+
+        if actualPath:
+            auto = Settings.get('mapGenerator/autostart', default=False, type=bool)
+            if not auto:
+                msgbox = QtWidgets.QMessageBox()
+                msgbox.setWindowTitle("Generate map")
+                msgbox.setText("Seems that you don't have the map used this game. Do you want to generate it?<br/><b>" + mapname + "</b>")
+                msgbox.setInformativeText("Map generation is CPU intensive task and may take some time.")
+                msgbox.setStandardButtons(QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.YesToAll | QtWidgets.QMessageBox.No)
+                result = msgbox.exec_()
+                if result == QtWidgets.QMessageBox.No:
+                    return False
+                elif result == QtWidgets.QMessageBox.YesToAll:
+                    Settings.set('mapGenerator/autostart', True)
+            
+            # Start generator with progress bar      
+            self.generatorProcess = MapGeneratorProcess(actualPath, self.mapsFolder, seed, version, mapname)
+            
+            # Check if map exists or generator failed
+            if os.path.isdir(os.path.join(self.mapsFolder, mapname)):
+                if self.previousMaps:
+                    self.previousMaps = self.previousMaps + ";" + mapname
+                else:
+                    self.previousMaps = mapname
+                Settings.set('mapGenerator/deleteMaps', self.previousMaps)
+                
+                return mapname
+            else:
+                return False
+        else:
+            return False
+            
+    def generateRandomMap(self):
+        ''' 
+        Called when user click "generate map" in host widget.
+        Prepares seed and requests latest version once per session
+        '''
+
+        if self.currentVersion == "0" or not self.latestVersion:
+            self.checkUpdates()
+
+            if self.latestVersion and self.versionController(self.latestVersion):
+                self.currentVersion = self.latestVersion   # mapgen is up-to-date
+                Settings.set('mapGenerator/version', self.currentVersion)
+            elif self.currentVersion == "0":               # if not "0", use older version
+                return False                               # otherwise we don't have any generator at all
+
+        seed = ''.join(["%s" % random.randint(0, 9) for num in range(0, 19)])
```

Comment:
This is a bad way to generate a random string of digits.
Also, the generator expects a 64-bit, signed seed as string.
This may go completely haywire if your 19-decimal seed exceeds the 64-bit range.

Reply:
It's good enough IMO. Just make it generate 8 bytes, then concatenate them as hex and we're good.

Reply:
That's not how it's supposed to work. Where's the trouble with generating a random 64 bit integer?

Reply:
Ok, I don't know why but when I use 
```random.getrandbits(64)```
or
```int.from_bytes(os.urandom(8), "big")```

they both freeze client on a second attempt (first is ok). While current rng always works smooth and never freezes

Reply:
NVM, freeze happens when cmd window has selection or smth. Not rng fault